plugin.video.montreal.greek-tv/resources/lib/control.py

```diff
@@ -0,0 +1,175 @@
+keyboard = xbmc.Keyboard
+sleep = xbmc.sleep
+execute = xbmc.executebuiltin
+skin = xbmc.getSkinDir()
+player = xbmc.Player()
+playlist = xbmc.PlayList(xbmc.PLAYLIST_VIDEO)
+monitor = xbmc.Monitor()
+wait = monitor.waitForAbort
+aborted = monitor.abortRequested
+
+transPath = xbmc.translatePath
+skinPath = xbmc.translatePath('special://skin/')
+addonPath = xbmc.translatePath(addonInfo('path'))
+legalfilename = xbmc.makeLegalFilename
+
+try:
+    dataPath = xbmc.translatePath(addonInfo('profile')).decode('utf-8')
+except AttributeError:
+    dataPath = xbmc.translatePath(addonInfo('profile'))
+
+window = xbmcgui.Window(10000)
+dialog = xbmcgui.Dialog()
+progressDialog = xbmcgui.DialogProgress()
+progressDialogGB = xbmcgui.DialogProgressBG()
+windowDialog = xbmcgui.WindowDialog()
+button = xbmcgui.ControlButton
+image = xbmcgui.ControlImage
+alphanum_input = xbmcgui.INPUT_ALPHANUM
+password_input = xbmcgui.INPUT_PASSWORD
+hide_input = xbmcgui.ALPHANUM_HIDE_INPUT
+verify = xbmcgui.PASSWORD_VERIFY
+item = xbmcgui.ListItem
+
+openFile = xbmcvfs.File
+makeFile = xbmcvfs.mkdir
+makeFiles = xbmcvfs.mkdirs
+deleteFile = xbmcvfs.delete
+deleteDir = xbmcvfs.rmdir
+listDir = xbmcvfs.listdir
+exists = xbmcvfs.exists
+copy = xbmcvfs.copy
+
+join = os.path.join
+settingsFile = os.path.join(dataPath, 'settings.xml')
+bookmarksFile = os.path.join(dataPath, 'bookmarks.db')
+cacheFile = os.path.join(dataPath, 'cache.db')
+
+
+def name():
+
+    return addonInfo('name')
+
+
+def version():
+
+    return addonInfo('version')
+
+
+def fanart():
+
+    return addonInfo('fanart')
+
+
+def infoDialog(message, heading=addonInfo('name'), icon='', time=3000):
+
+    if icon == '':
+        icon = addonInfo('icon')
+
+    try:
+
+        dialog.notification(heading, message, icon, time, sound=False)
+
+    except BaseException:
+
+        execute("Notification(%s, %s, %s, %s)" % (heading, message, time, icon))
+
+
+def okDialog(heading, line1):
+
+    return dialog.ok(heading, line1)
+
+
+def yesnoDialog(line1, line2='', line3='', heading=addonInfo('name'), nolabel=None, yeslabel=None):
+
+    return dialog.yesno(heading, line1, line2, line3, nolabel, yeslabel)
+
+
+def selectDialog(list, heading=addonInfo('name')):
+
+    return dialog.select(heading, list)
+
+
+def openSettings(query=None, id=addonInfo('id')):
+
+    try:
+
+        idle()
+        execute('Addon.OpenSettings({0})'.format(id))
+        if query is None:
+            raise Exception()
+        c, f = query.split('.')
+        execute('SetFocus(%i)' % (int(c) + 100))
+        execute('SetFocus(%i)' % (int(f) + 200))
+
+    except BaseException:
+
+        return
+
+
+# Alternative method
+def Settings(id=addonInfo('id')):
+
+    try:
+        idle()
+        addon(id).openSettings()
+    except BaseException:
+        return
+
+
+def refresh():
+
+    return execute('Container.Refresh')
+
+
+def idle():
+
+    return execute('Dialog.Close(busydialog)')
+
+
+def addonmedia(icon, addonid=addonInfo('id')):
```

Comment:
no need for this. `xbmcaddon.Addon().getAddonInfo("icon")` should return the icon.

Reply:
addonmedia function is used to simplify naming individual icons in directory module.